docs/website/docs/deployment-configurations/bare-metal.md

```diff
@@ -0,0 +1,108 @@
+
+## Prerequisites
+
+Out-of-tree bare-metal platform tools and source code for the system should be
+ready, such as
+
+* Compilation toolchain
+* Platform linker script
+* Firmware libraries
+
+Please follow the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#get-compiler-for-cpu-native-instructions)
+to retrieve the IREE compiler.
+
+## Compile the model for bare-metal
+
+The model can be compiled with the following command from the IREE compiler
+build directory
+
+``` shell hl_lines="3 4 5 6"
+iree/tools/iree-translate \
+    -iree-mlir-to-vm-bytecode-module \
+    -iree-hal-target-backends=dylib-llvm-aot \
+    -iree-llvm-link-embedded=true \
+    -iree-llvm-target-triple=x86_64-pc-linux-elf \
+    -iree-llvm-debug-symbols=false \
+    iree/samples/models/simple_abs.mlir \
+    -o /tmp/simple_abs_dylib.vmfb
+
+```
+
+In which
+
+* `iree-hal-target-backends=dylib-llvm-aot`: Build the model for the dynamic
+library CPU HAL driver
+* `iree-llvm-link-embedded=true`: Generate the dynamic library with
+[LLD](https://lld.llvm.org/) and the artifact can be loaded with the
+[embedded library loader](https://github.com/google/iree/blob/main/iree/hal/local/loaders/embedded_library_loader.h)
+without invoking the dynamic library support
+* `iree-llvm-target-triple`: Use the `<arch>-pc-linux-elf` LLVM target triple so
+the artifact has a fixed ABI to be rendered by the
+[elf_module library](https://github.com/google/iree/tree/main/iree/hal/local/elf)
+* `iree-llvm-debug-symbols=false`: To reduce the artifact size
+
+See [generate.sh](https://github.com/google/iree/blob/main/iree/hal/local/elf/testdata/generate.sh)
+for example command-line instructions of some common architectures
+
+You can replace the MLIR file with the other MLIR model files, following the
+[instructions](https://google.github.io/iree/deployment-configurations/cpu-dylib/#compile-the-model)
+
+!!! todo
+    add the static library support
+
+## Build bare-metal runtime from the source (CMake only)
+
+A few CMake options and macros should be set to build a subset of IREE runtime
+libraries compatible with the bare-metal platform. We assume there's no
+multi-thread control nor system library support in the bare-metal system. The
+model execution is in a single-thread synchronous fashion.
+
+### Set CMake options
+
+* `set(IREE_BUILD_COMPILER OFF)`: Build IREE runtime only
+* `set(IREE_ENABLE_MLIR OFF)`: Disable LLVM/MLIR dependencis for the runtime
+library
+* `set(CMAKE_SYSTEM_NAME Generic)`: Let CMake to generate the runtime without
+targeting a specific operating system
+* `set(IREE_BINDINGS_TFLITE OFF)`: Disable the TFLite binding support
+* `set(IREE_ENABLE_THREADINGS OFF)`: Disable multi-thread library support
+* `set(IREE_HAL_DRIVERS_TO_BUILD "Dylib;VMVX")`: Build the runtime HAL drivers
+for the dynamic library and VMVX backends.
+* `set(IREE_BUILD_TESTS OFF)`: IREE test tools assumes the multi-thread system
+support
```

Comment:
grammar/spelling nits and some rewording suggestions

```suggestion
* `set(IREE_BUILD_COMPILER OFF)`: Build IREE runtime only
* `set(IREE_ENABLE_MLIR OFF)`: Disable LLVM/MLIR dependencies for the runtime
library
* `set(CMAKE_SYSTEM_NAME Generic)`: Tell CMake to skip targeting a specific operating system
* `set(IREE_BINDINGS_TFLITE OFF)`: Disable the TFLite binding support
* `set(IREE_ENABLE_THREADING OFF)`: Disable multi-thread library support
* `set(IREE_HAL_DRIVERS_TO_BUILD "Dylib;VMVX")`: Build only the dynamic library and VMVX runtime HAL drivers
* `set(IREE_BUILD_TESTS OFF)`: Disable tests until IREE supports running them on bare-metal platforms
support
```

Reply:
Done.